### ISSUE TITLE
A few problems

### ISSUE THREAD

**Issue by UltimateCodeLove** [huludini/pw-web#43]
The panel doesn't allow you to login ingame when you register with Upper Case letters. It is possible to change this when you sign up, so either it tells the player to use lower case letters or automatically changes the username they entered into lower case ones?

I am not very familiar with the platform so I really couldn't find where this was controlled :/

Thank you!

**Comment by tomirons** [comment]
I'll guide you in the right direction to test this, if this works then I'll make the changes.

In `App/Http/Controllers/Auth/AuthController` change the following lines.

**139**: 
```
'name' => strtolower($data['name']),
```

**141**:
```
'passwd' => Hash::make(strtolower($data['name']).$data['password']),
```

Let me know if that works!

**Comment by UltimateCodeLove** [comment]
This works very well. Thank you!

If you are going to update the panel, can you please at least fix the Faction ranking please?

Thanks again!

**Comment by UltimateCodeLove** [comment]
Also the player ranking seems to be bugged, it doesn't show some of the players for some kind of reason.

My chat gets stuck at "Loading" even though all the settings are correct and the online number doesn't surpass 16 even though more are online than that.

Voucher doesn't work for quest activators :(

Is there any way to fix these?

Thank you!

**Comment by tomirons** [comment]
I'm going to need more details about the last two issues you're having. I've heard of the ranking not working, but that could be caused by a few things.

1. Scheduler isn't setup properly
2. Server configurations might not be correct
3. Ports might not be open.

I'll have to look at how the chat works, it's been awhile since I've messed with it.

What do you mean the voucher doesn't work for quest activators?

**EDIT:** As I expected, the chat viewer loads a file and decodes the chat. So if you don't have the path and permissions for it setup correctly, it won't work.

**Comment by UltimateCodeLove** [comment]
The ranking still excludes some players and I don't know why.

I am using Plesk and I have given PW-web folder root ownership with sh privileges in Plesk. It still doesn't work for some reason.

Also, is it possible to set up a system so PW-web does an IP check for people who have voted? Some players make multiple accounts and they farm tons of gold with fake voting. 

I am also trying to set up a function so players can change their genders through the panel. Can you help me with this?

I know I need help with so many things but I absolutely love this panel and its design.

Thanks again!

**Comment by tomirons** [comment]
The system automatically checks for the users IP address for each vote. If they're getting by on voting they must be using a VPN.

I'm not 100% sure what's going on with the ranking and why it's not showing some players, but unfortunately I don't have the time to look into it. Same thing goes with the gender change, you can look at the source yourself and create a new service. It's not that difficult.

**Comment by UltimateCodeLove** [comment]
Even I can vote multiple times on the same IP and the timer starts counting down for every time I vote.

**Comment by tomirons** [comment]
My apologies, I missed one critical point. So when someone votes, it checks the IP and the account ID. The reason behind that is so if multiple people are playing from the same IP then they're able to vote without any issues.

You're more than welcome to change the source and remove the ID check, it will then only let one vote per IP through.

Remove lines 30 and 39 in `app/VoteLog.php`: https://github.com/tomirons/pw-web/blob/master/app/VoteLog.php

**Comment by UltimateCodeLove** [comment]
This is what I was exactly looking for, thank you so much!

I managed to make my own Gender change function, I hope it doesn't get deleted when you make new updates.

As for Faction rankings, I couldn't get them to work so I just removed the links.

For some reason, I still couldn't get chat to work, it still bothers me a bit because your panel fully displays all chat logs from the file. I think I will have to find another way to get it to work.

There is also a password reset table in the SQL database, is there any use for this or is it abandoned? Like can people reset their passwords through email?
